### ISSUE TITLE
Blender Lights Export Review

### ISSUE THREAD

**Issue by trevorsandy** [leozide/leocad#914]
Hi Leo,

I've reviewed the Blender and POV-Ray lights behaviour up to commit [b6de368](https://github.com/leozide/leocad/commit/b6de368f6dd15029d399fc2621de19d4e160bae2) in order to implement the changes in [blenderldrawrender](https://github.com/trevorsandy/blenderldrawrender). There are some outstanding items I'd like to clarify and close before I commence with the updates. 

For reference here are the Blender light and light object transform defaults:
![Blender_Light_Defaults_01](https://github.com/leozide/leocad/assets/13388970/eb0335e1-0988-4527-9756-2eeebe80fed8)

![Blender_Light_Transform_Defaults_01](https://github.com/leozide/leocad/assets/13388970/d7ca8ebe-4266-4863-af26-1e13635edd63)

Here are my findings:

- [ ] General
	- [ ] POWER default value changed from _10_ to _1_ - Consider using Blender default _1000_ 
	- [ ] `ATTENUATION_DISTANCE` and `ATTENUATION_POWER` not expected by Blender
	- [ ] Change transform `ROTATION` format from _XYZ_ vectors to _XYZ_ Euler angles
	- [ ] When moved, `ROTATION` values in Properties do not match values in bottom-right UI message bar		
	- [ ] No UI to distinguish between POV-Ray and Blender lights. For Blender lights:
		- [ ] Exported `POWER` default is _1_ but Blender `POWER` default is _1000_
		- [ ] Pointlight `RADIUS` not exported, `SIZE` not expected
		- [ ] Spotlight `RADIUS` exported as first value of `SIZE` pair - e.g. `SIZE .2 0`
		- [ ] Directional light `ANGLE` not exported, `SIZE` not expected
		- [ ] Arealight `AREA_GRID` not expected
- [ ] Pointlight
	- [ ] `RADIUS` not exported
	- [ ] `RADIUS` default changed from _.25_ to _0_ - Consider using Blender default _.1_
	- [ ] `SIZE` not expected by Blender
- [ ] Spotlight
	- [ ] `SPOT_BLEND` missing
	- [ ] `RADIUS` exported as first value of `SIZE` pair - Consider exporting as `RADIUS` n
	- [ ] `RADIUS (SIZE)` default changed from _.25_ to _0_ - Consider using Blender default _.1_
	- [ ] Cannot enter more than _1_ decimal place for `RADIUS (SIZE)` property
- [ ] Directional (Sun) Light
	- [ ] `SIZE` not expected by Blender 
	- [ ] `ANGLE` not exported
	- [ ] `ANGLE` default changed from _11.4_ to _0.0002_ 
- [ ] Arealight
	- [ ] `AREA_GRID` not expected by Blender

It would be enough to simply add your response (will/won't change etc...) below the respective bullet item.

Cheers,

**Comment by leozide** [comment]
Hi Trevor,

I wanted to have a single light for both Blender and POV-Ray, and people can configure the app specific settings if they want to. I think this works better because in general people would place the same light for both apps and tune app-specific parameters if needed but I'm happy to discuss.

> * [ ]  General
>   
>   * [ ]  POWER default value changed from _10_ to _1_ - Consider using Blender default _1000_
Default power for me in Blender is 10

>   * [ ]  `ATTENUATION_DISTANCE` and `ATTENUATION_POWER` not expected by Blender
These are POV-Ray only (fade_distance and fade_power).

>   * [ ]  Change transform `ROTATION` format from _XYZ_ vectors to _XYZ_ Euler angles
This is a rotation matrix, not vectors. You can convert it to euler angles like we do on the UI, but I'd rather save it as a matrix.

>   * [ ]  When moved, `ROTATION` values in Properties do not match values in bottom-right UI message bar
We don't display the rotation on the status bar, what do you mean?

>   * [ ]  No UI to distinguish between POV-Ray and Blender lights. For Blender lights:
See first paragraph.

>     * [ ]  Exported `POWER` default is _1_ but Blender `POWER` default is _1000_
Default power for me in Blender is 10

>     * [ ]  Pointlight `RADIUS` not exported, `SIZE` not expected
>     * [ ]  Spotlight `RADIUS` exported as first value of `SIZE` pair - e.g. `SIZE .2 0`
Size is radius, I can change the exported key for clarity.

>     * [ ]  Directional light `ANGLE` not exported, `SIZE` not expected
Size is angle, I can change the exported key for clarity.

>     * [ ]  Arealight `AREA_GRID` not expected
This is POV-Ray only

> * [ ]  Pointlight
>   
>   * [ ]  `RADIUS` not exported
>   * [ ]  `RADIUS` default changed from _.25_ to _0_ - Consider using Blender default _.1_
Default radius in Blender is 0 for me

>   * [ ]  `SIZE` not expected by Blender
See above.

> * [ ]  Spotlight
>   * [ ]  `SPOT_BLEND` missing
SPOT_BLEND = SPOT_PENUMBRA_ANGLE / SPOT_CONE_ANGLE, I followed POV-Ray here because I thought it made more sense at the time, let me think about this one.

>   * [ ]  `RADIUS` exported as first value of `SIZE` pair - Consider exporting as `RADIUS` n
>   * [ ]  `RADIUS (SIZE)` default changed from _.25_ to _0_ - Consider using Blender default _.1_
Default size in Blender is 0 for me

>   * [ ]  Cannot enter more than _1_ decimal place for `RADIUS (SIZE)` property
I'll take a look.

> * [ ]  Directional (Sun) Light
>   * [ ]  `SIZE` not expected by Blender
>   * [ ]  `ANGLE` not exported
See above.

>   * [ ]  `ANGLE` default changed from _11.4_ to _0.0002_
Default angle in blender for me is 0.526

> * [ ]  Arealight
>   * [ ]  `AREA_GRID` not expected by Blender
See above.

I don't know why my defaults are different from yours (I have Blender 4.0), I don't mind changing them once we figure out what they should be.

I'll rename some of the keys for clarity.

**Comment by trevorsandy** [comment]
> I wanted to have a single light for both Blender and POV-Ray, and people can configure the app specific settings if they want to. I think this works better because in general people would place the same light for both apps and tune app-specific parameters if needed but I'm happy to discuss.

Indeed, and I agree. However, taking into account the different input parameter between POV-Ray and Blender, I felt it was necessary to distinguish between the two applications. To this end, I imagine one could consult the Parameter UI (as I did in the initial PR) or generate the appropriate indicator depending on the selected Render menu item selected. Nevertheless, I'll conclude that there won't be any behaviour to distinguish between POV-Ray and Blender.

> Default power for me in Blender is _10_.

Currently _10_ is also the default I expect from the input file in blenderldrawrender. So if you revert to using _10_, I would not have to make any changes.  

> These are POV-Ray only (fade_distance and fade_power).

Indeed. With no way to distinguish between POV-Ray and Blender, I imagine you are flushing parameters for both on save/export.

> `SPOT_BLEND` = `SPOT_PENUMBRA_ANGLE` / `SPOT_CONE_ANGLE`, I followed POV-Ray here because I thought it made more sense at the time, let me think about this one.

Very well. So as the default `SPOT_PENUMBRA_ANGLE` value is _0_ the default `SPOT_BLEND` will calculate to _0_ also.

> This is a rotation matrix, not vectors. You can convert it to euler angles like we do on the UI, but I'd rather save it as a matrix.

Very well. 

> I don't know why my defaults are different from yours (I have Blender 4.0), I don't mind changing them once we figure out what they should be.

I'm using Blender 3.4.1, but I installed 4.1.0 and also got the same defaults. Perhaps these defaults depend on screen-size but I could not find any definitive information about this.

Cheers,
